### PR TITLE
OSSM-5685 Support offline installation through default image digests

### PR DESCRIPTION
**Pull request by dgn**
This adds the support for defining default image digests in the operator config, similar to how it worked in 2.x - however, in this new implementation it is much easier for a user to overwrite an image/hub/tag for one component and still use the default images for other components.

### CODE REVIEW

File: controllers/istio/istio_controller_test.go [hunk 1]
```diff
@@ -1328,6 +1329,182 @@ func TestMergeOverwrite(t *testing.T) {
...
+	testCases := []struct {
+		name         string
+		config       common.OperatorConfig
+		inputIstio   *v1alpha1.Istio
+		inputValues  helm.HelmValues
+		expectValues helm.HelmValues
+	}{
+		{
+			name: "no-user-values",
+			config: common.OperatorConfig{
```
Comment (jwendell) [inline controllers/istio/istio_controller_test.go:1363]: can you add a test with no (or empty) `config`?

**Review by jwendell — COMMENTED** [review]
LGTM (just a nit), will defer to Marko. Let me know if you want me to approve it right away

File: controllers/istio/istio_controller_test.go (hunk 1 above)
Comment (dgn) [reply controllers/istio/istio_controller_test.go:1363]: done!

**Review by luksa — APPROVED** [review]
